gcloud/bigquery/dataset.py

```diff
@@ -204,6 +273,24 @@ def _require_client(self, client):
             client = self._client
         return client
 
+    def _parse_access_grants(self, access):
+        """Parse a resource fragment into a set of access grants.
+
+        :type access: list of mappings
+        :param access: each mapping represents a single access grant
+
+        :rtype: list of :class:`AccessGrant`
+        :returns: a list of parsed grants
+        """
+        result = []
+        for grant in access:
+            grant = grant.copy()
+            role = grant.pop('role')
+            entity_type, entity_id = list(grant.items())[0]
```

Comment:
@fhoffa Is it possible for there to be more than one key in an "access" (ACL) object returned?

The [spec][1] indicates

```json
    {
      "role": "...",
      "userByEmail": "...",
      "groupByEmail": "...",
      "domain": "...",
      "specialGroup": "...",
      "view": {
        "projectId": "...",
        "datasetId": "...",
        "tableId": "..."
      }
    }
```

but empirically @tseaver seems to only be seeing one key and it is always one of `userByEmail`, `groupByEmail` and `specialGroup`.

[1]: https://cloud.google.com/bigquery/docs/reference/v2/datasets